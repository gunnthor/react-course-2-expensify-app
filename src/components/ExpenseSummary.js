import React from 'react';
import { connect } from 'react-redux';
import selectTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral'

export const ExpenseSummary = ({total, expensesCount}) => {
  let oneProp = (
    <h1>Viewing 1 expense totalling {numeral(total/100).format('$0,0.00')}</h1>
  );
  let moreProps = (
    <h1>Viewing {expensesCount} expenses totalling {numeral(total/100).format('$0,0.00')} </h1>
  )
  return (
  <div>
    {expensesCount > 0 && (expensesCount === 1 ? <div>{oneProp}</div> : <div>{moreProps}</div>)}
  </div>
)}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    total: selectTotal(visibleExpenses),
    expensesCount: visibleExpenses.length
  };
}; 

export default connect(mapStateToProps)(ExpenseSummary);