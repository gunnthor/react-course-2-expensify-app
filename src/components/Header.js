import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <div className="headerStyle">
      <h1>Expensify</h1>
      <div className="linkerne">
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard </NavLink>  
        <NavLink to="/create" activeClassName="is-active" >Create Expense </NavLink>  
        <NavLink to="/help" activeClassName="is-active">Help Page </NavLink>  
        <button onClick={startLogout}> Logout </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);