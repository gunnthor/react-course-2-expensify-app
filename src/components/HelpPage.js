import React from 'react';

const HelpPage = () => (
  <div>
    This is the Help Page <b/>
    <p>How may i help?</p>
  </div>
);

export default HelpPage;