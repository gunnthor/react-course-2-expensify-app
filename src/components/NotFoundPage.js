import React from 'react';
import { Link } from 'react-router-dom';

export const NotFoundPage = () => (
  <div>
    404! - Page not found. <h4><Link to="/">Go Home</Link></h4>
  </div>
);

export default NotFoundPage