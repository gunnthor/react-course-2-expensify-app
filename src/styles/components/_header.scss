.headerStyle {
  margin-bottom: $m-size;
  background: gray;
}