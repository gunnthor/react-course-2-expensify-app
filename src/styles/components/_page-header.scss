.page-header {
  background: $off-white;
  margin-bottom: $l-size;
  padding: $l-size 0;
}

.page-header__actions {
  margin-top: $m-size;
}

.page-header__title {
  font-weight: 300;
  font-size: $l-size;
  span {
    font-weight: bold;
  }
}