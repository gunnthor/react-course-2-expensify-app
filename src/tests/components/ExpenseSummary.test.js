import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('Should correctly render expenseSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary total={235} expenses={1} />)
  expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenseSummary with multiple expense', () => {
  const wrapper = shallow(<ExpenseSummary total={123456} expenses={23} />)
  expect(wrapper).toMatchSnapshot();
});