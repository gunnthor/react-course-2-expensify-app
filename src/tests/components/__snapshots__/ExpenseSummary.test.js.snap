// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenseSummary with 1 expense 1`] = `<div />`;

exports[`Should correctly render expenseSummary with multiple expense 1`] = `<div />`;
